src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/ManagingOfficerAdditionService.java

```diff
@@ -0,0 +1,108 @@
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.PersonName;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.additions.Addition;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.additions.CorporateManagingOfficerAddition;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.additions.IndividualManagingOfficerAddition;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.commonmodels.CompanyIdentification;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import static uk.gov.companieshouse.overseasentitiesapi.utils.TypeConverter.addressDtoToAddress;
+
+@Service
+public class ManagingOfficerAdditionService {
+    public List<Addition> managingOfficerAdditions(OverseasEntitySubmissionDto overseasEntitySubmissionDto) {
+        List<Addition> additions = new ArrayList<>();
+
+        addIndividualManagingOfficerAdditions(additions, overseasEntitySubmissionDto);
+        addCorporateManagingOfficerAdditions(additions, overseasEntitySubmissionDto);
+
+        return additions;
+    }
+
+    private void addIndividualManagingOfficerAdditions(
+            List<Addition> additions, OverseasEntitySubmissionDto overseasEntitySubmissionDto) {
+        var managingOfficerIndividual = overseasEntitySubmissionDto.getManagingOfficersIndividual();
+        if (managingOfficerIndividual != null) {
+            managingOfficerIndividual.stream()
+                    .filter(managingOfficer -> (managingOfficer.getChipsReference() == null))
+                    .map(this::getIndividualManagingOfficerAddition)
+                    .forEachOrdered(additions::add);
+        }
+    }
+
+    private IndividualManagingOfficerAddition getIndividualManagingOfficerAddition(ManagingOfficerIndividualDto mo) {
+        var actionDate = mo.getStartDate();
+        var residentialAddress = mo.getUsualResidentialAddress();
+        var serviceAddress = mo.getServiceAddress();
+        var resignedOn = mo.getResignedOn();
+
+        var individualManagingOfficerAddition =
+                new IndividualManagingOfficerAddition(
+                        actionDate,
+                        addressDtoToAddress(residentialAddress),
+                        addressDtoToAddress(serviceAddress),
+                        resignedOn);
+
+        individualManagingOfficerAddition.setPersonName(new PersonName(mo.getFirstName(), mo.getLastName()));
+        individualManagingOfficerAddition.setFormerNames(mo.getFormerNames());
+        individualManagingOfficerAddition.setBirthDate(mo.getDateOfBirth());
+        individualManagingOfficerAddition.setRole(mo.getRoleAndResponsibilities());
+        individualManagingOfficerAddition.setOccupation(mo.getOccupation());
+
+        if (Objects.isNull(mo.getSecondNationality())) {
```

Comment:
We may be doing this in several places so worth pulling out a utility method? Just the test and optional concat.